app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -158,7 +164,8 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             SpeechProcessingIntentProcessor(this, components.core.store, components.analytics.metrics),
             StartSearchIntentProcessor(components.analytics.metrics),
             OpenBrowserIntentProcessor(this, ::getIntentSessionId),
-            OpenSpecificTabIntentProcessor(this)
+            OpenSpecificTabIntentProcessor(this),
+            DefaultBrowserIntentProcessor(this)
```

Comment:
Looking at the implementation of `DefaultBrowserIntentProcessor`, I don't see this touching IO but our intent processors increase, so our VIEW startup performance might be affected.

cc: @mcomella

Reply:
The processing of the intent is fairly quick.  I don't think it will affect the VIEW startup performance unless the intent is from the default browser notification.  It checks if the intents have the `INTENT_DEFAULT_BROWSER_NOTIFICATION` extra.  if not then it returns right away.

Reply:
In the event that we take the `true` branch of the if statement in that class, I'm also concerned this could impact start up perf:
```kotlin
        return if (isDefaultBrowserNotificationIntent(intent)) {
            activity.openSetDefaultBrowserOption()
            activity.settings().defaultBrowserNotificationDisplayed = true
```

- `openSetDefaultBrowserOption` appears to access a `SystemService`. The `SystemService` classes often rely on IPC which can be a little slow (e.g. this is why we're modifying how `isDefaultBrowser` is called in https://github.com/mozilla-mobile/fenix/pull/20116/)
- We write to settings and any access of shared prefs is has some perf impact

Perhaps you can measure how long this branch takes _on a low-end device_ like the Moto G5? e.g.
```kotlin
val before = SystemClock.elapsedRealtime()
return if (isDefaultBrowserNotificationIntent(intent)) {
            activity.openSetDefaultBrowserOption()
            activity.settings().defaultBrowserNotificationDisplayed = true
            true
...
}.also {
    val end = SystemClock.elapsedRealtime()
    Log.e("lol", "elapsed: ${end - start}")
}
```

If you don't have a low-end device, let me know and I can take the measurement if you send me a build. :)

Reply:
If the timing is 10ms+ on a Moto G5-equivalent device, we probably want to change it. Most code of this size should run in 0-1ms (i.e. 1 is a rounding situation).

Reply:
Yes, you're correct that in the event that the intent is coming from the default browser notification, it will have an affect on the startup time.

But this will only happen if the user taps on the notification.  Shouldn't we then perform user's action as soon as we can?

Reply:
rocketsroger on Slack:

> I’m not sure if we should worry about the impact to performance when the intent is coming from the notification.  Since it only happens once in user’s lifetime and only happens if the reason for starting Fenix was because the user tapped on the notification.

Ah, it only happens once? Yeah, probably not worth optimizing then. We should probably add a comment to that effect. Good catch!

Reply:
Sure, I'll add a comment  Thanks,

Reply:
Added comment in `DefaultBrowserIntentProcessor.kt`